tests/integration/deploying/deploying_contract_test.py

```diff
@@ -0,0 +1,57 @@
+# pylint: disable=consider-using-with
+import subprocess
+import time
+from distutils.file_util import copy_file
+from pathlib import Path
+from socket import socket as Socket
+
+import pytest
+
+from protostar.commands.deploy.deploy_contract import deploy_contract
+
+
+def get_available_port() -> int:
+    with Socket() as socket:
+        socket.bind(("", 0))
+        return socket.getsockname()[1]
+
+
+@pytest.fixture(name="devnet_gateway_url")
+def devnet_gateway_url_fixture():
+    devnet_port = get_available_port()
+
+    command = [
+        "poetry",
+        "run",
+        "starknet-devnet",
+        "--host",
+        "localhost",
+        "--port",
+        str(devnet_port),
+    ]
+    # pylint: disable=consider-using-with
+    proc = subprocess.Popen(command)
+    time.sleep(5)
+    yield f"http://localhost:{devnet_port}"
+    proc.kill()
+
+
+@pytest.fixture(name="compiled_contract_file")
+def compiled_contract_file_fixture(tmpdir):
+    copy_file(
+        str(Path(__file__).parent / "data" / "main.json"),
```

Comment:
I think we should use name `fixtures` not `data` as directory name to remain consistent.

Reply:
I think we should use https://pypi.org/project/pytest-datadir/. It uses `data` directory or the name of the module.